Match_The_Bracket.rb

```diff
@@ -0,0 +1,35 @@
+=begin
+  This problem was asked by Facebook.
+
+Given a string of round, curly, and square open and closing brackets, 
+return whether the brackets are balanced (well-formed).
+For example, given the string "([])", you should return true.
+Given the string "([)]" or "((()", you should return false.
+  
+=end
+$brackets = { "{" => "}", "[" => "]", "(" => ")" }
+
+def index_finder(string)
+  reference = ["{", "[", "("]
+  string.chars.each_with_index do |element, index|
+    return index - 1 if !$brackets.has_key?(element)
+  end
+end
+
+def bracket(string)

```

Comment:
can you example the algorithm here? 

I think you can do this much simpler way. why no just look at the first character and look at the last character and see if they are matching pair, then the second one and the second last one, etc. and you keep going that way till you meet in the middle. 

its true if there is a matching pair for all. you can return false early if there is not a matching bracket. This method will have a few edge cases but that would solve for all the examples that we have above.

Reply:
This algorithm is wrong, because this can't detect any bracket inside the brackets. Example {[]()}

Reply:
I have modified the algorithm like you said, which satisfies the question. But if i put this Example { [ ] ( ) } it will not work for it.